src/handlers/v1/markets/[chainId]/[marketId]/collateral-swap/index.ts

```diff
@@ -0,0 +1,150 @@
+import * as common from '@protocolink/common';
+import { utils } from 'ethers';
+import { validateMarket } from 'src/validations';
+
+type GetCollateralSwapQuotationRouteParams = EventPathParameters<{ chainId: string; marketId: string }> &
+  EventBody<{
+    account?: string;
+    withdrawalToken?: common.TokenObject;
+    amount?: string;
+    targetToken?: common.TokenObject;
+    slippage?: number;
+  }>;
+
+type GetCollateralSwapQuotationResponseBody = QuoteAPIResponseBody<CollateralSwapQuotation>;
+
+export const v1GetCollateralSwapQuotationRoute: Route<GetCollateralSwapQuotationRouteParams> = {
+  method: 'POST',
+  path: '/v1/markets/{chainId}/{marketId}/collateral-swap',
+  handler: async (event) => {
+    const chainId = Number(event.pathParameters.chainId);
+    const marketId = event.pathParameters.marketId.toUpperCase();
+    if (!validateMarket(chainId, marketId)) {
+      throw newHttpError(400, { code: '400.1', message: 'market does not exist' });
+    }
+
+    if (!event.body) {
+      throw newHttpError(400, { code: '400.2', message: 'body is invalid' });
+    }
+
+    let account = event.body.account;
+    if (!account) {
+      throw newHttpError(400, { code: '400.3', message: `account can't be blank` });
+    }
+    try {
+      account = utils.getAddress(account);
+    } catch {
+      throw newHttpError(400, { code: '400.4', message: 'account is invalid' });
+    }
+
+    const service = new Service(chainId);
+
+    let marketInfo: MarketInfo;
+    try {
+      marketInfo = await service.getMarketInfo(marketId, account);
+    } catch (err) {
+      throw newInternalServerError(err);
+    }
+    const { utilization, healthRate, netAPR, borrowUSD } = marketInfo;
+    const currentPosition = { utilization, healthRate, netAPR, totalDebt: borrowUSD };
+
+    let targetAmount = '0';
+    const logics: GetCollateralSwapQuotationResponseBody['logics'] = [];
+    let approvals: GetCollateralSwapQuotationResponseBody['approvals'] = [];
+    let targetPosition = currentPosition;
+    if (event.body.withdrawalToken && event.body.targetToken && event.body.amount && Number(event.body.amount) > 0) {
+      const { withdrawalToken, targetToken, amount, slippage } = event.body;
+      const { supplyAPR, supplyUSD, borrowAPR, collateralUSD, borrowCapacityUSD, liquidationLimit, collaterals } =
+        marketInfo;
+
+      // Verify token input
+      const _withdrawalToken = common.Token.from(withdrawalToken);
+      const withdrawalCollateral = collaterals.find(({ asset }) => asset.is(_withdrawalToken.unwrapped));
+      if (!withdrawalCollateral) {
+        throw newHttpError(400, { code: '400.5', message: 'withdrawal token is not collateral' });
+      }
+      const _targetToken = common.Token.from(targetToken);
+      const targetCollateral = collaterals.find(({ asset }) => asset.is(_targetToken.unwrapped));
+      if (!targetCollateral) {
+        throw newHttpError(400, { code: '400.5', message: 'target token is not collateral' });
```

Comment:
Should add an unit test.
```suggestion
        throw newHttpError(400, { code: '400.6', message: 'target token is not collateral' });
```